### PR TITLE
[Unity][Training] Loss functions and AppendLoss pass

### PR DESCRIPTION
**Pull request by Ubospica**
This PR adds two components to support the Relax training workflow: loss function library and AppendLoss pass.
- Loss function library. A library that generates relax functions representing common loss functions, such as MSELoss and CrossEntropyLoss. 
  - Generally, loss function will take one or more **input parameters** (that is outputs of the backbone of a model), one or more **target parameters**, and generate a scalar value denoting the loss.
- AppendLoss pass. You can specify one function (the backbone function) in the IRModule, and one loss function (normally, generated by the loss function library) to AppendLoss. It will attach the loss function to the backbone function, and add the result function to the IRModule.

Co-authored-by: Chaofan Lin <siriusneo@sjtu.edu.cn>

### CODE REVIEW

**Review by SiriusNEO — APPROVED** [review]

**Review by MasterJH5574 — APPROVED** [review]

**Comment by MasterJH5574** [comment]
Please resolve the conflict and we’re good to go :-)

**Comment by Ubospica** [comment]
@MasterJH5574 We can go now :)
